File: cdm/cpp/patient/actions/SEAcuteRespiratoryDistressSyndromeExacerbation.cpp
/* Distributed under the Apache License, Version 2.0.
   See accompanying NOTICE file for details.*/

#include "stdafx.h"
#include "patient/actions/SEAcuteRespiratoryDistressSyndromeExacerbation.h"
#include "properties/SEScalar0To1.h"
#include "io/protobuf/PBPatientActions.h"

SEAcuteRespiratoryDistressSyndromeExacerbation::SEAcuteRespiratoryDistressSyndromeExacerbation() : SEPatientAction()
{
  m_Severity=nullptr;
}

SEAcuteRespiratoryDistressSyndromeExacerbation::~SEAcuteRespiratoryDistressSyndromeExacerbation()
{
  Clear();
}

void SEAcuteRespiratoryDistressSyndromeExacerbation::Clear()
{
  SEPatientAction::Clear();
  SAFE_DELETE(m_Severity);
}

void SEAcuteRespiratoryDistressSyndromeExacerbation::Copy(const SEAcuteRespiratoryDistressSyndromeExacerbation& src)
{
  PBPatientAction::Copy(src, *this);
}

bool SEAcuteRespiratoryDistressSyndromeExacerbation::IsValid() const
{
  return HasSeverity();
}
bool SEAcuteRespiratoryDistressSyndromeExacerbation::IsActive() const
{
  if (!IsValid())
    return false;
  return GetSeverity() > 0;
}

bool SEAcuteRespiratoryDistressSyndromeExacerbation::HasSeverity() const
{
  return m_Severity==nullptr?false:m_Severity->IsValid();
}

SEScalar0To1& SEAcuteRespiratoryDistressSyndromeExacerbation::GetSeverity()
{
  if(m_Severity==nullptr)
    m_Severity=new SEScalar0To1();
  return *m_Severity;
}
double SEAcuteRespiratoryDistressSyndromeExacerbation::GetSeverity() const
{
  if (m_Severity == nullptr)
    return SEScalar::dNaN();
  return m_Severity->GetValue();
}

void SEAcuteRespiratoryDistressSyndromeExacerbation::ToString(std::ostream &str) const
{
  str << "Patient Action : ARDS Exacerbation"; 
  if(HasComment())
    str<<"\n\tComment: "<<m_Comment;
  str << "\n\tSeverity: "; HasSeverity() ? str << m_Severity : str << "NaN";
  str << std::flush;
}

File: cdm/csharp/patient/actions/SEAcuteRespiratoryDistressSyndromeExacerbation.cs
/* Distributed under the Apache License, Version 2.0.
   See accompanying NOTICE file for details.*/

namespace Pulse.CDM
{
  public class SEAcuteRespiratoryDistressSyndromeExacerbation : SEPatientAction
  {
    protected SEScalar0To1 severity;
    protected SEScalar0To1 left_lung_affected;
    protected SEScalar0To1 right_lung_affected;

    public SEAcuteRespiratoryDistressSyndromeExacerbation()
    {
      severity = null;
      left_lung_affected = null;
      right_lung_affected = null;
    }

    public override void Clear()
    {
      base.Clear();
      if (severity != null)
        severity.Invalidate();
      if (left_lung_affected != null)
        left_lung_affected.Invalidate();
      if (right_lung_affected != null)
        right_lung_affected.Invalidate();
    }

    public override bool IsValid()
    {
      return HasSeverity() && HasLeftLungAffected() && HasRightLungAffected();
    }

    public bool HasSeverity()
    {
      return severity == null ? false : severity.IsValid();
    }
    public SEScalar0To1 GetSeverity()
    {
      if (severity == null)
        severity = new SEScalar0To1();
      return severity;
    }

    public bool HasLeftLungAffected()
    {
      return left_lung_affected == null ? false : left_lung_affected.IsValid();
    }
    public SEScalar0To1 GetLeftLungAffected()
    {
      if (left_lung_affected == null)
        left_lung_affected = new SEScalar0To1();
      return left_lung_affected;
    }

    public bool HasRightLungAffected()
    {
      return right_lung_affected == null ? false : right_lung_affected.IsValid();
    }
    public SEScalar0To1 GetRightLungAffected()
    {
      if (right_lung_affected == null)
        right_lung_affected = new SEScalar0To1();
      return right_lung_affected;
    }
  }
}
